core/shared/src/main/scala/zio/internal/FiberRuntime.scala

```diff
@@ -492,7 +508,7 @@ final class FiberRuntime[E, A](fiberId: FiberId.Runtime, fiberRefs0: FiberRefs,
    */
   private[zio] def getChildren(): JavaSet[Fiber.Runtime[_, _]] = {
     if (_children eq null) {
-      _children = Platform.newConcurrentWeakSet[Fiber.Runtime[_, _]]()(Unsafe.unsafe)
+      _children = Platform. /*newConcurrentWeakSet*/ newWeakSet[Fiber.Runtime[_, _]]()(Unsafe.unsafe)
```

Comment:
Nice. A bit dangerous, but nice. As a sanity check, can we triple check that no other fiber can write into `_children`?